code/modules/projectiles/projectile.dm

```diff
@@ -72,6 +72,10 @@
 	var/matrix/effect_transform			// matrix to rotate and scale projectile effects - putting it here so it doesn't
 										//  have to be recreated multiple times
 
+/obj/item/projectile/is_hot()
+	if (damage_type == BURN)
+		return damage*heat
```

Comment:
this will produce completely insane values, best not. Just go with 3800 as a flat value for lasers